python/makeALPACAEvents.py

```diff
@@ -120,25 +130,82 @@ def ntupleWrite(ctau,fn,Lmin,Lmax,startZ,endZ,SmearBeam,mres,gax,old):
         ntup.Fill(int(i),int(2),int(dau2[1]),float(dau2[7]),float(dau2[8]),float(dau2[9]),daux,dauy,dauz,int(0),float(1),float(dau2[10]),float(dau2[15])/10./c_light,w)
     ntup.Write()
     f.Close()
+    print("ALP -> Gamma + Gamma events are produced with cross-section of {} pb at mass of {} GeV and photon coupling of {} GeV^-1".format(xs,mres,gax))
     return inputFile
 
 def runEvents(mres,gax,nev,Lmin,Lmax,startZ,endZ,SmearBeam):
+    mres = float(mres)
+    gax  = float(gax)
     print('ALPACA is starting for mass of {} GeV with photon coupling coeffiecient {} GeV^-1.'.format(mres,gax))
-    pdg = ROOT.TDatabasePDG.Instance()
-    pdg.AddParticle('a','ALP', mres, False, gax, 0., 'a', 9900015)
+    addALPtoPDG(float(mres),float(gax))
     wdir = os.environ['ALPACABIN']
     old=os.getcwd()
     os.chdir(wdir)
     inputWrite(mres,gax,nev,Lmin,Lmax)
-    rn="./alpaca < input.DAT"
+    rn="./alpaca < "+"input"+naming+".DAT"
     os.system(rn)
     print('ALPACA generated the events.')
-    pa="./evrecs/evrec_"+str(mres)+"_"+str(gax)+".dat"
+    pa="./evrecs/evrec"+naming+".dat"
     fn=open(pa,"r")
```

Comment:
1. fn is never closed. I would suggest using a context manager again (`with open(pa, "r") as fn:`.

2. Also, please avoid names such as `fn`, `pa`, `rn`. They are meaningless to anyone but you (and even you might forget what they mean), and make the code harder to follow because of that.
Exceptions to this rule of thumb would be names conventionally used (e.g. because of the mathematical meaning).